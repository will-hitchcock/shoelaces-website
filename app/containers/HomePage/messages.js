/*
 * NotFoundPage Messages
 *
 * This contains all the text for the NotFoundPage component.
 */
import { defineMessages } from 'react-intl'

export default defineMessages({
  header: {
    id: 'app.components.HomePage.header',
    defaultMessage: '{ ðŸ‘Ÿ } CSS-in-JS flexbox grid'
  },
  docs: {
    id: 'app.components.HomePage.header',
    defaultMessage: 'Docs'
  },
  github: {
    id: 'app.components.HomePage.header',
    defaultMessage: 'Github'
  }
})
