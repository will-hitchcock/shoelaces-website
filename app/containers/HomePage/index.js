/*
 * HomePage
 *
 * This is the first thing users see of our App, at the '/' route
 *
 * NOTE: while this component should technically be a stateless functional
 * component (SFC), hot reloading does not currently support SFCs. If hot
 * reloading is not a necessity for you then you can refactor it and remove
 * the linting exception.
 */

import React, { PureComponent } from 'react'
import { FormattedMessage } from 'react-intl'
import { Container, Row, Col } from 'react-shoelaces'
import Button from 'components/Button'
import Box from 'components/Box'
import Header from 'components/Header'
import messages from './messages'

export default class HomePage extends PureComponent { // eslint-disable-line react/prefer-stateless-function
  render() {
    return (
      <Container>
        <Row>
          <Header>
            <FormattedMessage {...messages.header} />
          </Header>
        </Row>
        <Row>
          <Col xs="auto">
            <Button path="docs">Docs</Button>
          </Col>
          <Col xs="auto">
            <Button external path="https://github.com/will-hitchcock/shoelaces">
              Github
            </Button>
          </Col>
        </Row>
        <Row>
          <Col xs={4 / 12}><Box /></Col>
          <Col xs={4 / 12}><Box /></Col>
          <Col xs={4 / 12}><Box /></Col>
        </Row>
        <Row>
          <Col xs={6 / 12}><Box /></Col>
          <Col xs={6 / 12}><Box /></Col>
        </Row>
      </Container>
    )
  }
}
