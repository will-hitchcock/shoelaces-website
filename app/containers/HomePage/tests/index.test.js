import React from 'react'
import HomePage from '../index'

describe('<HomePage />', () => {
  it('renders the <HomePage />', () => {
    expect(shallow(<HomePage />)).toMatchSnapshot()
  })
})
