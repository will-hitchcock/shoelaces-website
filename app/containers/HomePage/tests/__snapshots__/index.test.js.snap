exports[`<HomePage /> renders the <HomePage /> 1`] = `
<styled.div>
  <styled.div>
    <Header
      type="h1">
      <FormattedMessage
        defaultMessage="{ ðŸ‘Ÿ } CSS-in-JS flexbox grid"
        id="app.components.HomePage.header"
        tagName="span"
        values={Object {}} />
    </Header>
  </styled.div>
  <styled.div>
    <styled.div
      xs="auto">
      <Button
        path="docs">
        Docs
      </Button>
    </styled.div>
    <styled.div
      xs="auto">
      <Button
        external={true}
        path="https://github.com/will-hitchcock/shoelaces">
        Github
      </Button>
    </styled.div>
  </styled.div>
  <styled.div>
    <styled.div
      xs={0.3333333333333333}>
      <styled.div />
    </styled.div>
    <styled.div
      xs={0.3333333333333333}>
      <styled.div />
    </styled.div>
    <styled.div
      xs={0.3333333333333333}>
      <styled.div />
    </styled.div>
  </styled.div>
  <styled.div>
    <styled.div
      xs={0.5}>
      <styled.div />
    </styled.div>
    <styled.div
      xs={0.5}>
      <styled.div />
    </styled.div>
  </styled.div>
</styled.div>
`;
